### PR TITLE
feat: add support for key type k256

### PR DESCRIPTION
**Pull request by sairanjit**
### Summary

- Added support for new KeyType - `k256`
- Added support for `did:key` for the curve - `secp256k1`

### CODE REVIEW

File: packages/core/src/crypto/jose/jwa/crv.ts [hunk 1]
```diff
@@ -4,4 +4,5 @@ export enum JwaCurve {
   P521 = 'P-521',
   Ed25519 = 'Ed25519',
   X25519 = 'X25519',
+  SECP256K1 = 'secp256k1',
```
Comment (TimoGlastra) [inline packages/core/src/crypto/jose/jwa/crv.ts:7]:
```suggestion
  Secp256k1 = 'secp256k1',
```

File: packages/core/src/crypto/jose/jwk/K256Jwk.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import { JwaCurve, JwaKeyType } from '../jwa'
+import { JwaSignatureAlgorithm } from '../jwa/alg'
+
+import { Jwk } from './Jwk'
+import { compress, expand } from './ecCompression'
+import { hasKty, hasCrv, hasX, hasY, hasValidUse } from './validate'
+
+export class K256Jwk extends Jwk {
+  public static readonly supportedEncryptionAlgorithms: JwaEncryptionAlgorithm[] = []
+  public static readonly supportedSignatureAlgorithms: JwaSignatureAlgorithm[] = [JwaSignatureAlgorithm.ES256]
```
Comment (TimoGlastra) [inline packages/core/src/crypto/jose/jwk/K256Jwk.ts:15]: This should be `ES256K` right?

```suggestion
  public static readonly supportedSignatureAlgorithms: JwaSignatureAlgorithm[] = [JwaSignatureAlgorithm. ES256K]
```

File: packages/core/src/crypto/jose/jwk/ecCompression.ts [hunk 3]
```diff
@@ -43,6 +44,21 @@ function getConstantsForCurve(curve: 'P-256' | 'P-384' | 'P-521') {
     pIdent = prime.add(1).divide(4)
   }
 
+  if (curve === 'secp256k1') {
```
Comment (TimoGlastra) [inline packages/core/src/crypto/jose/jwk/ecCompression.ts:47]: Curious, where did you find this? Can you add a link?

File: packages/core/src/crypto/jose/jwk/ecCompression.ts [hunk 4]
```diff
@@ -11,9 +11,10 @@ const curveToPointLength = {
   'P-256': 64,
```
Comment (TimoGlastra) [inline packages/core/src/crypto/jose/jwk/ecCompression.ts:11]: I think same here maybe. Would be good to use JwaCurve values here maybe?

**Review by TimoGlastra — APPROVED** [review]
Very nice!!! :)

File: packages/core/src/crypto/jose/jwa/crv.ts (hunk 1 above)
Comment (sairanjit) [reply packages/core/src/crypto/jose/jwa/crv.ts]: Updated value

File: packages/core/src/crypto/jose/jwk/K256Jwk.ts (hunk 2 above)
Comment (sairanjit) [reply packages/core/src/crypto/jose/jwk/K256Jwk.ts]: Good catch :)

File: packages/core/src/crypto/jose/jwk/ecCompression.ts (hunk 4 above)
Comment (sairanjit) [reply packages/core/src/crypto/jose/jwk/ecCompression.ts]: Added JwaCurve values

File: packages/core/src/crypto/jose/jwk/ecCompression.ts (hunk 3 above)
Comment (sairanjit) [reply packages/core/src/crypto/jose/jwk/ecCompression.ts]: Added a link [https://en.bitcoin.it/wiki/Secp256k1](https://en.bitcoin.it/wiki/Secp256k1)

**Comment by TimoGlastra** [comment]
Do the tests run locally for you @sairanjit? It keeps failing, and I'm not sure if it's related to your changes or not

**Comment by TimoGlastra** [comment]
Ah I merged #1629 and now there's some merge conflicts. Can you address it?

File: packages/askar/src/utils/askarKeyTypes.ts [hunk 5]
```diff
@@ -7,15 +7,38 @@ export enum AskarKeyTypePurpose {
...
+    purposes: [AskarKeyTypePurpose.KeyManagement],
+  },
+  [KeyType.P256]: {
+    keyAlg: KeyAlgs.EcSecp256r1,
+    purposes: [AskarKeyTypePurpose.KeyManagement],
+  },
+  [KeyType.K256]: {
+    keyAlg: KeyAlgs.EcSecp256k1,
+    purposes: [AskarKeyTypePurpose.KeyManagement],
+  },
```
Comment (TimoGlastra) [inline packages/askar/src/utils/askarKeyTypes.ts:37]: Wait I messed up I think. We should support signing and key management for the P/K 256

Comment (TimoGlastra) [reply packages/askar/src/utils/askarKeyTypes.ts:37]: Can you address it for both?

Comment (sairanjit) [reply packages/askar/src/utils/askarKeyTypes.ts:37]: Sure I will add them

**Pull request merged**